src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -73,73 +71,83 @@ internal partial class CodeFixService : ICodeFixService
+            // Hence, for Normal or Low priority, we only need to execute analyzers which can report at least one
+            // fixable diagnostic that can have a non-suppression/configuration fix. Note that for
+            // 'CodeActionPriorityRequest.High', we only run compiler analyzer, which always has fixable
+            // diagnostics, so we can pass in null. 
+
+            if (!(priority is CodeActionRequestPriority.Normal or CodeActionRequestPriority.Low))
+                return null;
+
+            var hasWorkspaceFixers = TryGetWorkspaceFixersMap(document, out var workspaceFixersMap);
+            var projectFixersMap = GetProjectFixers(document.Project);
 
-            if (!document.IsOpen())
+            return id =>
             {
-                return default;
-            }
+                if (hasWorkspaceFixers && workspaceFixersMap!.Value.ContainsKey(id))
+                    return true;
 
-            using var _ = ArrayBuilder<DiagnosticData>.GetInstance(out var diagnostics);
-            using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                return projectFixersMap.ContainsKey(id);
+            };
+        }
+
+        public async Task<FirstFixResult> GetMostSevereFixAsync(
+            Document document, TextSpan range, CodeActionRequestPriority priority, CodeActionOptions options, CancellationToken cancellationToken)
+        {
+            var (allDiagnostics, upToDate) = await _diagnosticService.TryGetDiagnosticsForSpanAsync(
+                document, range, GetShouldIncludeDiagnosticPredicate(document, priority),
+                includeSuppressedDiagnostics: false, priority, cancellationToken).ConfigureAwait(false);
+
+            var spanToDiagnostics = ConvertToMap(allDiagnostics);
 
+            using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
             var linkedToken = linkedTokenSource.Token;
 
-            // This flag is used by SuggestedActionsSource to track what solution is was
-            // last able to get "full results" for.
-            var isFullResult = await _diagnosticService.TryAppendDiagnosticsForSpanAsync(
-                document, range, diagnostics, cancellationToken: linkedToken).ConfigureAwait(false);
+            var spanToErrorDiagnostics = new SortedDictionary<TextSpan, List<DiagnosticData>>();
+            var spanToOtherDiagnostics = new SortedDictionary<TextSpan, List<DiagnosticData>>();
 
-            var errorDiagnostics = diagnostics.Where(d => d.Severity == DiagnosticSeverity.Error);
-            var otherDiagnostics = diagnostics.Where(d => d.Severity != DiagnosticSeverity.Error);
+            foreach (var (span, diagnostics) in spanToDiagnostics)
+            {
+                foreach (var diagnostic in diagnostics)
+                {
+                    var otherMap = diagnostic.Severity == DiagnosticSeverity.Error
```

Comment:
Hmm I would have assumed "otherMap" was the "the map that doesn't match the severity", but I guess not? Or is this a typo?

Reply:
(just call it "map"?)

Reply:
yes.  that name was terrible. changed to `preferredMap`.